app/api/ada/adaLocalStorage.js

```diff
@@ -23,11 +37,29 @@ const storageKeys = {
 export function saveCryptoAccount(
   ca: CryptoAccount
 ): void {
-  _saveInStorage(storageKeys.ACCOUNT_KEY, ca);
+  const localAccount: LocalStorageCryptoAccount = {
+    account: ca.account,
+    root_cached_key: ca.root_cached_key.key().to_hex(),
+    derivation_scheme: ca.derivation_scheme
+  };
+  _saveInStorage(storageKeys.ACCOUNT_KEY, localAccount);
 }
 
 export function getSingleCryptoAccount(): CryptoAccount {
-  return _getFromStorage(storageKeys.ACCOUNT_KEY);
+  const localAccount = _getFromStorage(storageKeys.ACCOUNT_KEY);
+  if (localAccount.derivation_scheme !== 'V2') {
+    throw Error('Sanity check');
+  }
+  const pubKey = RustModule.Wallet.PublicKey.from_hex(localAccount.root_cached_key);
+  const account = RustModule.Wallet.Bip44AccountPublic.new(
+    pubKey,
+    RustModule.Wallet.DerivationScheme.v2()
+  );
+  return {
+    account: localAccount.account,
+    root_cached_key: account,
+    derivation_scheme: localAccount.derivation_scheme
+  };
```

Comment:
Is `account` type here `string` or `Bip44AccountPublic`?

Reply:
`Bip44AccountPublic`. You can tell both from the return type and the way the `account` variable is generated.